packages/did-provider-ion/src/functions.ts

```diff
@@ -48,9 +55,75 @@ export const toIonPrivateKeyJwk = (privateKeyHex: string): JwkEs256k => {
  * @return The JWK
  */
 export const toIonPublicKeyJwk = (publicKeyHex: string): JwkEs256k => {
-  return toJwkEs256k(secp256k1KeyUtils.publicKeyJwkFromPublicKeyHex(publicKeyHex))
+  return toJwkEs256k(publicKeyJwkFromPublicKeyHex(publicKeyHex))
 }
 
+/**
+ * Example
+ * ```js
+ * {
+ *  kty: 'EC',
+ *  crv: 'secp256k1',
+ *  d: 'rhYFsBPF9q3-uZThy7B3c4LDF_8wnozFUAEm5LLC4Zw',
+ *  x: 'dWCvM4fTdeM0KmloF57zxtBPXTOythHPMm1HCLrdd3A',
+ *  y: '36uMVGM7hnw-N6GnjFcihWE3SkrhMLzzLCdPMXPEXlA',
+ *  kid: 'JUvpllMEYUZ2joO59UNui_XYDqxVqiFLLAJ8klWuPBw'
+ * }
+ * ```
+ * See [rfc7638](https://tools.ietf.org/html/rfc7638) for more details on Jwk.
+ */
+export const getKid = (
+  jwk: ISecp256k1PrivateKeyJwk | ISecp256k1PublicKeyJwk
+) => {
+  const copy = { ...jwk } as any;
+  delete copy.d;
+  delete copy.kid;
+  delete copy.alg;
+  const digest = crypto
+    .createHash('sha256')
+    .update(canonicalize(copy)!, 'utf-8')
+    .digest();
+
+  return base64url.encode(Buffer.from(digest));
+};
+
+/** convert compressed hex encoded private key to jwk */
+export const privateKeyJwkFromPrivateKeyHex = (privateKeyHex: string) => {
+  const jwk = {
+    ...keyto.from(privateKeyHex, 'blk').toJwk('private'),
+    crv: 'secp256k1',
+  };
+  const kid = getKid(jwk);
+  return {
+    ...jwk,
+    kid,
+  };
+};
+
+/** convert compressed hex encoded public key to jwk */
+export const publicKeyJwkFromPublicKeyHex = (publicKeyHex: string) => {
+  let key = publicKeyHex;
+  const compressedHexEncodedPublicKeyLength = 66;
+  if (publicKeyHex.length === compressedHexEncodedPublicKeyLength) {
+    const keyBin = secp256k1.publicKeyConvert(
+      Buffer.from(publicKeyHex, 'hex'),
+      false
+    );
+    key = Buffer.from(keyBin).toString('hex');
```

Comment:
We already have @ethersproject/* in the dependency tree (usually) so It would be better to use that for public key conversions instead of a new dependency.

Using this import `import { computePublicKey } from '@ethersproject/signing-key'`,
I think this would work to convert the key to uncompressed hex encoding:
```suggestion
    key = computePublicKey(`0x${publicKeyHex}`, false).substring(2)
```

This also removes the need for `Buffer` which requires special treatment on web.

Reply:
Removed Buffer